frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -270,81 +261,121 @@ const StorageClassEncryptionLabel: React.FC = () => {
+              onChange={setChecked}
+              className="ocs-storage-class-encryption__form-checkbox"
+              data-test="storage-class-encryption"
+            />
+            <span className="help-block">
+              {t(
+                'ceph-storage-plugin~An encryption key will be generated for each persistent volume created using this StorageClass.',
+              )}
             </span>
-          </TextListItem>
-          {currentKMS?.VAULT_CACERT && (
-            <TextListItem>
-              {t('ceph-storage-plugin~CA certificate:')}{' '}
-              <span className="help-block ocs-storage-class-encryption__help-block">
-                {t('ceph-storage-plugin~Provided')}
-              </span>
-            </TextListItem>
-          )}
-        </TextList>
-      </TextContent>
+          </FormGroup>
+        </Form>
+      </div>
+    )
+  );
+};
+
+const ExistingKMSDropDown: React.FC<ExistingKMSDropDownProps> = ({
+  csiConfigMap,
+  serviceName,
+  setEncryptionId,
+}) => {
+  const { t } = useTranslation();
+
+  const [isOpen, setOpen] = React.useState(false);
+
+  const handleDropdownChange = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    const encryptionId = e.currentTarget.id;
+    setEncryptionId(encryptionId);
+  };
+  const kmsDropdownItems = _.map(csiConfigMap?.data, (connectionDetails, connectionName) => {
+    const kmsData = JSON.parse(connectionDetails);
+    return (
+      <DropdownItem
+        key={connectionName}
+        component="button"
+        id={connectionName}
+        data-test={connectionName}
+        onClick={handleDropdownChange}
+        description={kmsData?.VAULT_ADDR}
+      >
+        {connectionName}
+      </DropdownItem>
+    );
+  });
+
+  return (
+    <div className="form-group">
+      <Dropdown
+        className="dropdown dropdown--full-width"
+        toggle={
+          <DropdownToggle
+            id="kms-dropdown-id"
+            data-test="kms-dropdown-toggle"
+            onToggle={() => setOpen(!isOpen)}
+            toggleIndicator={CaretDownIcon}
+          >
+            {serviceName || t('ceph-storage-plugin~Select an existing connection')}
+          </DropdownToggle>
+        }
+        isOpen={isOpen}
+        dropdownItems={kmsDropdownItems}
+        onSelect={() => setOpen(false)}
+        id="kms-connection"
+      />
     </div>
   );
 };
 
-type KMSDetailsProps = {
-  currentKMS: KMSConfigMap;
-  setEditKMS: React.Dispatch<React.SetStateAction<boolean>>;
+type ExistingKMSDropDownProps = {
+  csiConfigMap: ConfigMapKind;
+  serviceName: string;
+  setEncryptionId: (encryptionId: string) => void;
 };
 
-export const StorageClassEncryption: React.FC<ProvisionerProps> = ({
+export const StorageClassEncryptionKMSID: React.FC<ProvisionerProps> = ({
   parameterKey,
   onParamChange,
 }) => {
   const { t } = useTranslation();
+
   const isKmsSupported = useFlag(GUARDED_FEATURES.OCS_KMS);
```

Comment:
you already handled this is "StorageClassEncryption". if Enable Encryption not displayed then by default StorageClassEncryptionKMSID won't be displayed